### PR TITLE
Add datadog address field to support unix domain socket

### PR DESCRIPTION
**Pull request by chotiwat**
## PR Summary

This PR adds an `address` field (env: `DATADOG_ADDRESS`) that allows the user to send metrics to a datadog agent either via UDP at `host:port` or unix domain socket at `unix:///path/to/socket`.

 - **Type:** Features
 - **Issue Link:** N/A
 - **Intended Change Level:** patch

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

**Comment by dvdliao** [comment]
strange how sentry pkg test failing

**Comment by wcharczuk** [comment]
i have to merge my GRPC tracing stuff, let me do that first.

File: datadog/config.go [hunk 1]
```diff
@@ -67,6 +69,14 @@ func (c Config) TracingEnabledOrDefault() bool {
 	return DefaultTracingEnabled
 }
 
+// GetAddress returns the datadog collector address string.
+func (c Config) GetAddress() string {
+	if c.Address != "" {
+		return c.Address
+	}
+	return c.GetHost()
```
Comment (wcharczuk) [inline datadog/config.go]: this is conflating the two fields, we should join the host and the port to create a fully formed URI, you will also probably need some helpers for sniffing for `unix://` on the prefix of the address but will depend on how the dd stuff works

File: datadog/config.go [hunk 2]
```diff
@@ -14,6 +14,8 @@ const (
 
 // Config is the datadog config.
 type Config struct {
+	// Address is the address of the datadog collector in the form of "hostname:port" or "unix:///path/to/socket"
+	Address string `json:"address,omitempty" yaml:"address,omitempty" env:"DATADOG_ADDRESS"`
```
Comment (dhermes) [inline datadog/config.go]: Why wouldn't it be in the form of `{protocol}://{hostname}:{port}`?

File: datadog/config.go (hunk 1 above)
Comment (dhermes) [reply datadog/config.go]: I'd imagine `statsd.New()` is following Postel's law and being lenient if the caller passes host+port without a protocol.

File: datadog/config.go (hunk 2 above)
Comment (wcharczuk) [reply datadog/config.go]: for networked collectors the protocol is UDP so it's inferred and not part of a proto in a uri.

Comment (dhermes) [reply datadog/config.go]: Ah duh, me missing the whole point of this PR 😁

**Pull request merged**